File: 0138/138.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var copyListWithRandomPointer = function(root) {
    if (!root) return 0;
    return 1 + Math.max(copyListWithRandomPointer(root.left), copyListWithRandomPointer(root.right));
};

File: 0138/138.go
func copyListWithRandomPointer(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := copyListWithRandomPointer(root.Left)
	r := copyListWithRandomPointer(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}